src/Task.cc

```diff
@@ -3694,6 +3694,73 @@ static void copy_mem_mapping(Task* from, Task* to, const KernelMapping& km) {
   }
 }
 
+static bool copy_mem_mapping_just_used(Task* from, Task* to, const KernelMapping& km)
+{
+  ScopedFd& fd = from->pagemap_fd();
+  if (!fd.is_open()) {
+    LOG(debug) << "Failed to open " << from->proc_pagemap_path();
+    return false;
+  }
+
+  size_t pagesize = page_size();
+
+  vector<uint64_t> buf;
+#if defined(__i386__)
+  const auto buf_page_count = 64*1024;
```

Comment:
Let's just use 64K for all architectures. That's still a 256K buffer. I think allocating a temporary 8MB buffer here is overkill.